### DISCUSSION TITLE
how to use Ex3_FFarmCompleteSetup.py

### DISCUSSION THREAD

**Discussion by clun1** [OpenFAST/python-toolbox#80] [General]
Hello,
 
I am trying to use Ex3_FFarmCompleteSetup.py, but I don't know how to set the path in it. Is there any information to explain it? or teach me how to use this case? thank you!

![image](https://github.com/OpenFAST/python-toolbox/assets/93849049/717b00fa-a7c8-4013-bd34-11685a57ce09)
![image](https://github.com/OpenFAST/python-toolbox/assets/93849049/b7c324c8-5f09-4c98-bff3-beb21f8fdd02)
![image](https://github.com/OpenFAST/python-toolbox/assets/93849049/ef94d1c7-aa82-4af7-b8a1-f9fd5da3add5)
![image](https://github.com/OpenFAST/python-toolbox/assets/93849049/2cfa7a41-683f-4372-8372-552900c3f391)

**Comment by ebranlard** [comment]
I think there are a couple of mistakes, the `templatePath` should be a directory.  All the files listed below should preferably be in this directory. Your `libdisconpath` should be the path to the `dll` or `.so` file.  SLURM is the system used in our cluster so you don't need to worry about it for now.  I'm not sure about the turbsim paths, @rthedin can help. 

It is likely that we will update this example in the future (I'd like it to be partially tested by our continuous-integration suite).

**Comment by rthedin** [comment]
As indicated by the commented-out lines, `templatePath` should be a directory, and `libdisconfilepath` should be the full path to the controller shared object file. In fact, [an error is raised](https://github.com/OpenFAST/python-toolbox/blob/48956911b2628fd570391f9eb8f52be63d5f19b6/pyFAST/fastfarm/FASTFarmCaseCreation.py#L818-L819) when you don't give an actual path to `templatePath`, with a message saying that the path doesn't exist. Similarly, [another error is raised](https://github.com/OpenFAST/python-toolbox/blob/48956911b2628fd570391f9eb8f52be63d5f19b6/pyFAST/fastfarm/FASTFarmCaseCreation.py#L914-L915) if `libdisconfilepath` isn't a shared object. I realize that is hard-coded for `.so` files only, so you will need to modify it to use your `dll` if you are on a Windows machine.

The TurbSim files are given in the [`SampleFiles`](https://github.com/OpenFAST/python-toolbox/tree/main/pyFAST/fastfarm/examples/SampleFiles) directory, included in this repository.

**Comment by ebranlard** [comment]
pyFAST has been renamed to `openfast_toolbox` and migrated here: https://github.com/OpenFAST/openfast_toolbox . 

Please continue the discussion in this new repository. Sorry for the inconvenience.
